cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -144,78 +144,27 @@ func Run(gadgetRoot, device string, options Options) error {
 		}
 	}
 
-	// store the encryption key as the last part of the process to reduce the
-	// possibility of exiting with an error after the TPM provisioning
-	if options.Encrypt {
-		if err := tpmSealKey(key, rkey, options); err != nil {
-			return fmt.Errorf("cannot seal the encryption key: %v", err)
-		}
+	if !options.Encrypt {
+		return nil
 	}
 
-	return nil
-}
-
-func tpmSealKey(key partition.EncryptionKey, rkey partition.RecoveryKey, options Options) error {
-	if err := rkey.Store(options.RecoveryKeyFile); err != nil {
+	// Write the recovery key
+	if err := rkey.Save(options.RecoveryKeyFile); err != nil {
 		return fmt.Errorf("cannot store recovery key: %v", err)
 	}
 
-	tpm, err := secboot.NewTPMSupport()
-	if err != nil {
-		return fmt.Errorf("cannot initialize TPM: %v", err)
-	}
-
-	if options.TPMLockoutAuthFile != "" {
-		if err := tpm.StoreLockoutAuth(options.TPMLockoutAuthFile); err != nil {
-			return fmt.Errorf("cannot store TPM lockout authorization: %v", err)
-		}
-	}
-
-	shim := filepath.Join(boot.InitramfsUbuntuBootDir, "EFI/boot/bootx64.efi")
-	grub := filepath.Join(boot.InitramfsUbuntuBootDir, "EFI/boot/grubx64.efi")
-
-	// TODO:UC20: Fix EFI image loading events
-	//
-	// The idea of EFIImageLoadEvent is to build a set of load paths for the current
-	// device configuration. So you could have something like this:
-	//
-	// shim -> recovery grub -> recovery kernel 1
-	//                      |-> recovery kernel 2
-	//                      |-> recovery kernel ...
-	//                      |-> normal grub -> run kernel good
-	//                                     |-> run kernel try
-	//
-	// Or it could look like this, which is the same thing:
-	//
-	// shim -> recovery grub -> recovery kernel 1
-	// shim -> recovery grub -> recovery kernel 2
-	// shim -> recovery grub -> recovery kernel ...
-	// shim -> recovery grub -> normal grub -> run kernel good
-	// shim -> recovery grub -> normal grub -> run kernel try
-	//
-	// This implementation in #8476, seems to just build a tree of shim -> grub -> kernel
-	// sequences for every combination of supplied input file, although the code here just
-	// specifies a single shim, grub and kernel binary, so you get one load path that looks
-	// like this:
-	//
-	// shim -> grub -> kernel
-	//
-	// This is ok for now because every boot path uses the same chain of trust, regardless
-	// of which kernel you're booting or whether you're booting through both the recovery
-	// and normal grubs. But when we add the ability to seal against specific binaries in
-	// order to secure the system with the Microsoft chain of trust, then the actual trees
-	// of EFIImageLoadEvents will need to match the exact supported boot sequences.
-
```

Comment:
this comment seems to be lost, but AFAICT we still have the same limitation? Can you re-add this comment somewhere?

Reply:
Will re-add the part mentioning the Microsoft chain of trust. I'll also need to clarify some points regarding the boot sequences with Chris, since we're already using the Microsoft certificate but still not defining all the exact boot sequences for different modes.